### DISCUSSION TITLE
Having problems loading a ".nc" file

### DISCUSSION THREAD

**Discussion by AvitayGT** [pydata/xarray#5105] [Q&A] [answered]
Hello,

I have a dataset of .nc format which contains the following variables:

![image](https://user-images.githubusercontent.com/81641993/113483099-6b457f80-94aa-11eb-8801-dfb226de67c0.png)

When trying to load the file, I get the following error.
What can be done to avoid this error and to successfully load the dataset? I guess I need to change the variable name somehow so it doesn't share the same name with one of its dimensions anymore, but I don't know how to gain access to the dataset's variables' names.

![image](https://user-images.githubusercontent.com/81641993/113483212-f6bf1080-94aa-11eb-9dcc-40d094ce8b44.png)

**Comment by keewis** [comment, marked as answer]
unfortunately, this is not possible with `xarray`, yet (see #2368). You will have to use the `netCDF4` library (or any other netcdf reading library, or the `ncrename` executable) to rename the conflicting variables:
```python
In [1]: import netCDF4
   ...: import xarray as xr
   ...: 
   ...: # create the file
   ...: with netCDF4.Dataset("test.nc", mode="w") as nc:
   ...:     nc.createDimension("time", size=1)
   ...:     nc.createDimension("lat", size=2)
   ...:     nc.createDimension("lon", size=2)
   ...: 
   ...:     nc.createVariable("lon", "float64", dimensions=("time", "lat", "lon"))
   ...:     nc.variables["lon"][:] = [[[0, 1], [2, 3]]]
   ...: 
   ...: # fails
   ...: #xr.open_dataset("test.nc")
   ...: 
   ...: # rename variables
   ...: with netCDF4.Dataset("test.nc", mode="r+") as nc:
   ...:     display(nc)
   ...: 
   ...:     nc.renameVariable("lon", "lon_")
   ...:     display(nc)
   ...: 
   ...: with xr.open_dataset("test.nc") as ds:
   ...:     display(ds)
<class 'netCDF4._netCDF4.Dataset'>
root group (NETCDF4 data model, file format HDF5):
    dimensions(sizes): time(1), lat(2), lon(2)
    variables(dimensions): float64 lon(time, lat, lon)
    groups: 
<class 'netCDF4._netCDF4.Dataset'>
root group (NETCDF4 data model, file format HDF5):
    dimensions(sizes): time(1), lat(2), lon(2)
    variables(dimensions): float64 lon_(time, lat, lon)
    groups: 
<xarray.Dataset>
Dimensions:  (lat: 2, lon_: 2, time: 1)
Dimensions without coordinates: lat, lon_, time
Data variables:
    lon      (time, lat, lon_) float64 ...
```

**Reply by AvitayGT** [reply]
Thank you very much! You helped me a lot.
